pkg/controllers/provisioning/provisioner.go

```diff
@@ -142,7 +134,10 @@ func (p *Provisioner) isProvisionable(ctx context.Context, candidate *v1.Pod) (b
 	return !pod.IsScheduled(stored), nil
 }
 
-func (p *Provisioner) launch(ctx context.Context, constraints *v1alpha5.Constraints, packing *binpacking.Packing) error {
+func (p *Provisioner) launch(ctx context.Context, node *scheduling.Node) error {
+	if len(node.InstanceTypeOptions) == 0 {
```

Comment:
Can't we stop this from being zero in the packing logic?  If instance types goes to zero, we fail the packing.

Reply:
I could check for this earlier, but this happens if you do something like request 1Tb of memory.  No node has it, we create a new node, it ends up with zero instance types after the pod is added and then nothing is compatible with it in the future.

Reply:
> create a new node

Does it make sense for this to fail since the Add should fail? Maybe we should decouple NewNode and Add and check compatibility of the first pod.

Reply:
sgtm